### ISSUE TITLE
radio_setup(); issue - openiboot.c line 89

### ISSUE THREAD

**Issue by Anon0** [planetbeing/iphonelinux#3]
After some digging and more digging I have been able to compile the ToolChain and open iBoot as a pre-compiled version just flat out didn't load. Anyways, I "sudo loadibec openiboot.img3" and everything seems to go ok. I get the openiboot software loaded and I then try to boot the main os and that just flashes a white screen and loops back into the openiboot loader. ... I select the other option and then it partially loads. Something may be going wrong earlier in the loading process making it impossible for the rest of the tool to load.

iPhone 3G Model: MB702LL
Firmware Version: 3.1.2 ( 7D11 )
iBoot Version: Haven't Checked
JailbreakSW: blackra1n ( No mac to pwn it with )
UnlockSW: sn0w
OS: Ubuntu 9.10 x86 ( Stock Install with all necessary tools & libs installed )

After comparing my openiboot output to http://pastie.org/pastes/929164

i2c: ack not received before byte 0 <- "Linked to Installing gamma table 0x0000c223 / 0x0000ff77"

radio: comm board detected. <- Where mine stops
-> Next? USB state change: 1 -> 3

------------------------ OPENIBOOT OUTPUT BEGIN ------------------------------
miu: epoch mismatch
PLL 1: off.
PLL 0: off.
PLL 1: 412000000
PLL 2: off.
PLL 3: 108000000
NOR vendor=bf, device=8e
Successfully loaded bank1 nvram
Successfully loaded bank2 nvram
syrah_quiesce()
otf clock divisor: 5
fps set to: 59.976
SFN: 0x10300, Addr: 0xfd00000, Size: 0x14001e0, hspan: 0x280, QLEN: 0xa0
syrah_init() -- Hurray for displays
Syrah Panel ID (0xb3c22b):
   Build:          UNKNOWN 
   Type:           TMD 
   Project/Driver: N82/Novatek-6.x AutoBoot Enabled 
Autoboot display...
changing to continuous calibration mode 3
syrah_init success!
Installing gamma table 0x0000c223 / 0x0000ff77
i2c: ack not received before byte 0 <- Repeats a bunch of times, edited for space
camera: sensor model ID = 0x1580
menu: images loaded
USB state change: 0 -> 0
EP 0: 2
EP 1: 2
EP 2: 2
EP 3: 2
EP 4: 2
EP 5: 2
USB state change: 0 -> 1
usb: reset detected
USB state change: 1 -> 1
usb: reset detected
USB state change: 1 -> 1
USB state change: 1 -> 3
USB state change: 3 -> 4
radio: comm board detected.
--------------------------- END OPENIBOOT OUTPUT ------------------------------------

Anyways, instead of griping about it I will dig through the code and see if I can work around it. Keep up the good work!!

-- ADDITIONAL DETAILS --
After recompiling in debug mode and commenting out radio_setup(); in openiboot.c on line 89 it loads completely! Alright, good! I successfully created the norbackup.dump .... so the issue is with radio_setup();

miu: epoch mismatch
PLL 1: off.
PLL 0: off.
PLL 1: 412000000
PLL 2: off.
PLL 3: 108000000
NOR vendor=bf, device=8e
Successfully loaded bank1 nvram
Successfully loaded bank2 nvram
syrah_quiesce()
otf clock divisor: 5
fps set to: 59.976
SFN: 0x10300, Addr: 0xfd00000, Size: 0x14001e0, hspan: 0x280, QLEN: 0xa0
syrah_init() -- Hurray for displays
Syrah Panel ID (0xb3c22b):
   Build:          UNKNOWN 
   Type:           TMD 
   Project/Driver: N82/Novatek-6.x AutoBoot Enabled 
Autoboot display...
changing to continuous calibration mode 3
syrah_init success!
Installing gamma table 0x0000c223 / 0x0000ff77
i2c: ack not received before byte 0 <- Repeats a bunch of times, edited for space
camera: sensor model ID = 0x1580
menu: images loaded
USB state change: 0 -> 0
EP 0: 2
EP 1: 2
EP 2: 2
EP 3: 2
EP 4: 2
EP 5: 2
USB state change: 0 -> 1
sdio: selecting voltage index 23
sdio: cccr version: 0, sdio version: 0, low-speed: 0, high-speed: 0, wide bus: 1, multi-block: 1, functions: 1
sdio: turning off pull-up resistor on DAT[3]
Manufacturer ID: 0x2df, product ID: 0x9103
Function: 0, max block size: 256
Function: 1, max block size: 256, enable timeout: 1000 ms
sdio: Ready!
sdio: enabled function 1
wlan: ioport = 0x10000
wlan: firmware status = 0x0
multitouch: powering on
usb: reset detected
USB state change: 1 -> 1
usb: reset detected
USB state change: 1 -> 1
i2c: ack not received before byte 0
i2c: ack not received before byte 0
i2c: ack not received before register/command 0
als: error initializing
nand: Probing flash controller...
USB state change: 1 -> 3
USB state change: 3 -> 4
nand: DEVICE: b614d5ad
nand: BANKS_TOTAL: 4
nand: BLOCKS_PER_BANK: 4096
nand: SUBLKS_TOTAL: 4096
nand: USER_SUBLKS_TOTAL: 3872
nand: PAGES_PER_SUBLK: 512
nand: PAGES_PER_BANK: 524288
nand: SECTORS_PER_PAGE: 8
nand: BYTES_PER_SPARE: 128
nand: BYTES_PER_PAGE: 4096
nand: PAGES_PER_BLOCK: 128
ftl: Attempting to read 128 pages from first block of first bank.
ftl: Found production format: 43303035
ftl: Successfully found FTL context block: 2570
ftl: Successfully read FTL context block. usnDec = 0xfffbf080
ftl: Detected version 46560001 b9a9fffe
ftl: FTL successfully opened!
bdev: partition id: 0, type: af, sectors: 63 - 128063
## bdev: partition id: 1, type: af, sectors: 128079 - 1982421

```
          WELCOME TO OPENIBOOT
```

---

```
                DEBUG MODE
```

i2c: ack not received before byte 0
nor_read 0x09000000 0x0 1048576
Reading 0x0 - 0x100000 to 0x9000000...
Done.
~norbackup.dump:1048576
file sent.

**Comment by dnf-xx** [comment]
May be not related to this ticket, but when I boot linux kernel, oibc stops output on line: "radio: reading baseband nvram...". On device screen output continues with "done ... bla-bla-bla" and kernel boot starts.

**Comment by ghost** [comment]
Yup - I found this issue as well with my iPhone 3G.  OpeniBoot will currently function on the iPhone 3G _only_ if radio_setup() is not called.  As stated above, the current workaround is to comment out line 89 of openiboot.c and recompile the openiboot.img3.  Second confirmation that it works with that hack.

UPDATE: Newest updates to git repository fix these issues with the radio init.

**Comment by Anon0** [comment]
Ok, the newest update does fix the issue of the initial openiboot run; however there is still an issue with the radio. Openiboot hangs/doesn't respond on a 3G when you try to register your radio on a cellular network or try to make a call. I would like to contribute to this; however, all I have been able to do is decrypt openiboot and haven't a clue as where to start "reversing" it. I compiled eda-2 but still havn't a clue as how to use it yet ... or even sure if that is what this is for.
